File: rust/src/models/list_checkout_sessions_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::checkout_session::CheckoutSession;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ListCheckoutSessionsResponse {
    pub sessions: Vec<CheckoutSession>,
}

impl ListCheckoutSessionsResponse {
    pub fn new(sessions: Vec<CheckoutSession>) -> Self {
        Self { sessions }
    }
}

File: typescript/src/models/listCheckoutSessionsResponse.ts
// this file is @generated
import { type CheckoutSession, CheckoutSessionSerializer } from "./checkoutSession";

export interface ListCheckoutSessionsResponse {
  sessions: CheckoutSession[];
}

export const ListCheckoutSessionsResponseSerializer = {
  _fromJsonObject(object: any): ListCheckoutSessionsResponse {
    return {
      sessions: object["sessions"].map((item: any) =>
        CheckoutSessionSerializer._fromJsonObject(item)
      ),
    };
  },

  _toJsonObject(self: ListCheckoutSessionsResponse): any {
    return {
      sessions: self.sessions.map((item: any) =>
        CheckoutSessionSerializer._toJsonObject(item)
      ),
    };
  },
};
